### PR TITLE
First draft at expanded README file - to describe goals and usage of operator

### PR DESCRIPTION
**Pull request by ldimaggi**
At some point we may want to expand this further into a formal operator user guide document. As a first step, we should describe the goals of the operator and how it works for prospective users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,121 @@
...
+    - description: Database connection port
+      displayName: DB port
+      path: dbConnectionPort
+      x-descriptors:
+    - urn:alm:descriptor:servicebindingrequest:env:attribute
+```
+
+## Quick Start
+
+To build:
```
Comment (otaviof) [inline README.md]: Should we point the audience to run Makefile targets instead? All the `kubectl apply -f` commands are covered after PR #52. :-)

File: README.md [hunk 2]
```diff
@@ -1 +1,121 @@
...
+* The app author (developer) creates a ServiceBindingRequest and specifies a label in the form of "connects-to." For example, to bind to an operator-backed database service created using a CR named "my-db", the label will be "connects-to=my-db".
+
+* The Service Binding Operator then:
+  * Reads backing service operator OLM descriptor to discover the binding attributes
+  * Creates a binding secret for the database
+  * Injects environment variables into the applications's DeploymentConfig, Deployment, Replicaset
+
+Here is an example of the "bind-able" operator OLM Descriptor - in ths case for a PostgreSQL database backing operator:
+
+```
```
Comment (otaviof) [inline README.md]: After the triple backtick you can add a ` yaml` (space followed by the format), therefore GitHub will render colorized syntax. For instance:
```
yaml:
  entry: value
  int: 1
```
Versus:
``` yaml
yaml:
  entry: value
  int: 1
```

File: README.md (hunk 1 above)
Comment (pratikjagrut) [reply README.md]: Yes right, we have make targets for basic purposes and we can add more as we get to know other purposes. So it’s good idea to point audience towards makefile target. Maybe in this process @ldimaggi could also identify new targets which can be part of makefile.

File: README.md [hunk 3]
```diff
@@ -1 +1,121 @@
...
+To run the operator locally:
+
+	kubectl apply -f deploy/service_account.yaml
+	kubectl apply -f deploy/role.yaml
+	kubectl apply -f deploy/role_binding.yaml
+	kubectl apply -f deploy/crds/apps_v1alpha1_servicebindingrequest_crd.yaml
+
+	And then you can: 
+	operator-sdk up local --verbose --namespace default
+
```
Comment (pratikjagrut) [inline README.md]: To run operator locally we have ‘make local’ target you can use it.

**Comment by baijum** [comment]
Please wrap each line at 80 characters.
Ref: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#general-file-guidelines

**Comment by sbose78** [comment]
Since the `lint` is happy, I'll go ahead and merge this so that we have a README which describes what it does. Let's keep adding more to the docs.

**Pull request merged**